lib/drives/models/recurring_drive.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+import 'package:rrule/rrule.dart';
+
+import '../../account/models/profile.dart';
+import '../../util/parse_helper.dart';
+import '../../util/search/position.dart';
+import '../../util/supabase_manager.dart';
+import '../../util/trip/trip_like.dart';
+import 'drive.dart';
+
+class RecurringDrive extends TripLike {
+  DateTime? stoppedAt;
+  RecurrenceRule recurrenceRule;
+
+  final int driverId;
+  final Profile? driver;
+
+  final TimeOfDay startTime;
+  final TimeOfDay endTime;
+
+  final List<Drive>? drives;
+
+  RecurringDrive({
+    super.id,
+    super.createdAt,
+    required super.start,
+    required super.startPosition,
+    required this.startTime,
+    required super.end,
+    required super.endPosition,
+    required this.endTime,
+    required super.seats,
+    this.stoppedAt,
+    required this.recurrenceRule,
+    required this.driverId,
+    this.driver,
+    this.drives,
+  });
+
+  @override
+  factory RecurringDrive.fromJson(Map<String, dynamic> json) {
+    return RecurringDrive(
+      id: json['id'] as int,
+      createdAt: DateTime.parse(json['created_at'] as String),
+      start: json['start'] as String,
+      startPosition: Position.fromDynamicValues(json['start_lat'], json['start_lng']),
+      startTime: TimeOfDay.fromDateTime(DateFormat('hh:mm:ss').parse(json['start_time'] as String)),
+      end: json['end'] as String,
+      endPosition: Position.fromDynamicValues(json['end_lat'], json['end_lng']),
+      endTime: TimeOfDay.fromDateTime(DateFormat('hh:mm:ss').parse(json['end_time'] as String)),
+      seats: json['seats'] as int,
+      stoppedAt: json['stopped_at'] == null ? null : DateTime.parse(json['stopped_at'] as String),
+      recurrenceRule: RecurrenceRule.fromString((json['recurrence_rule'] as String).split('\n')[1]),
+      driverId: json['driver_id'] as int,
+      driver: json.containsKey('driver') ? Profile.fromJson(json['driver'] as Map<String, dynamic>) : null,
+      drives: json.containsKey('drives') ? Drive.fromJsonList(parseHelper.parseListOfMaps(json['drives'])) : null,
+    );
+  }
+
+  static List<RecurringDrive> fromJsonList(List<Map<String, dynamic>> jsonList) {
+    return jsonList.map((Map<String, dynamic> json) => RecurringDrive.fromJson(json)).toList();
+  }
+
+  @override
+  Map<String, dynamic> toJson() {
+    return super.toJson()
+      ..addAll(<String, dynamic>{
+        'stopped_at': stoppedAt?.toString(),
+        'start_time': startTime.formatted,
+        'end_time': endTime.formatted,
+        'recurrence_rule': 'DTSTART:${(createdAt ?? DateTime.now()).millisecondsSinceEpoch}\n$recurrenceRule',
```

Comment:
Here we discussed to allow shenanigans like pausing a recurrence rule by setting the start time to something else right? I still don't know if within the recurrence rule is the right place to do that or if it should just be another column

Reply:
I'll have to think about that some more. But I don't think that should be part of the first PR for this feature, it's more of an auxiliary feature building on that one.